src/Cardano/AddressDerivation/Byron.hs

```diff
@@ -0,0 +1,242 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RoleAnnotations #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- Implementation of address derivation for the random scheme, as
+-- implemented by the legacy Cardano wallets.
+--
+-- For full documentation of the key derivation schemes,
+-- see the "Cardano.Crypto.Wallet" module, and the implementation in
+-- <https://github.com/input-output-hk/cardano-crypto/blob/4590efa638397e952a51a8994b5543e4ea3c1ecd/cbits/encrypted_sign.c cardano-crypto>.
+
+module Cardano.AddressDerivation.Byron
+    ( -- * Types
+      Byron(..)
+
+      -- * Generation
+    , unsafeGenerateKeyFromSeed
+    , generateKeyFromSeed
+    , minSeedLengthBytes
+    , unsafeMkByronKeyFromMasterKey
+    , mkByronKeyFromMasterKey
+
+      -- * Derivation
+    , deriveAccountPrivateKeyImpl
+    , deriveAddressPrivateKeyImpl
```

Comment:
We may review this a bit later but I don't think we should export these two functions. That's why we have the `HardDerivation` interface.

Reply:
indeed, the same can be applied to `generateFromSeed` as now it is part of `GenMasterKey` interface